src/state/axisMapSlice.ts

```diff
@@ -12,7 +13,18 @@ export type AxisId = string | number;
 export type AxisSettings = {
   id: AxisId;
   scale: ScaleType | RechartsScale | undefined;
-  type: 'number' | 'category';
+  type: AxisDomainType;
+  padding: XAxisPadding;
+  domain: AxisDomain | undefined;
+  allowDataOverflow: boolean;
+  /**
+   * The axis functionality is severely restricted without a dataKey
+   * - but there is still something left, and the prop is optional
+   * so this can also be undefined even in real charts.
+   * There are no defaults.
+   */
+  dataKey: DataKey<any> | undefined;
```

Comment:
Don't axes grab onto specific dataKeys under certain conditions? Like if you specific on a graphical item then Y axis is defaulted. Vice versa for X axis if layout is swapped

Reply:
Indeed, graphical items have their own axis maps. That comes next